File: Aigilas/Aigilas/States/GameplayState.cs
﻿using SPX.States;
using SPX.Entities;
using Aigilas.Creatures;
using Aigilas.Dungeons;
using SPX.Core;
using System;

namespace Aigilas.States
{
    public class GameplayState : State
    {
        public GameplayState()
        {
            EntityManager.Reset();
            CreatureFactory.ResetPlayerCount();
            DungeonFactory.Start();
            Console.WriteLine(Input.GetPlayerCount());
        }
        public void Update()
        {
            EntityManager.Update();
        }
        public void LoadContent()
        {
            EntityManager.LoadContent();
        }
        public void Draw()
        {
            EntityManager.Draw();
        }
    }
}

File: proj/src/aigilas/states/GameplayState.java
package aigilas.states;

import spx.entities.EntityManager;
import spx.net.Client;
import spx.states.State;
import aigilas.dungeons.DungeonFactory;

public class GameplayState implements State {
	public GameplayState() {
		System.out.println("Generating the dungeon...");
		EntityManager.Reset();
		DungeonFactory.Start();
		Client.Get().DungeonHasLoaded();
	}

	@Override
	public void Update() {
		EntityManager.Update();
	}

	@Override
	public void LoadContent() {
		EntityManager.LoadContent();
	}

	@Override
	public void Draw() {
		EntityManager.Draw();
	}
}
